### DISCUSSION TITLE
How to build project without EAS ?

### DISCUSSION THREAD

**Discussion by mars1211** [infinitered/ignite#2690] [General]
I'm trying to use your boiler plate, but I can't figure out how to build an android and ios app offline?

I think it's not quite right to link the project to a cloud service, especially a paid one! It's very repulsive.

Help create a workaround!

**Comment by morganick** [comment]
Hey @mars1211 all of the following scripts inside of `package.json` are local:

```
    "build:ios:sim": "eas build --profile development --platform ios --local",
    "build:ios:dev": "eas build --profile development:device --platform ios --local",
    "build:ios:prod": "eas build --profile production --platform ios --local",
    "build:android:sim": "eas build --profile development --platform android --local",
    "build:android:dev": "eas build --profile development:device --platform android --local",
    "build:android:prod": "eas build --profile production --platform android --local",
    "start": "npx expo start",
    "android": "npx expo run:android",
    "ios": "npx expo run:ios",
```

You can use `eas` without using their cloud offering. We use it because it simplifies building your application. You can read more about building locally here: https://docs.expo.dev/build-reference/local-builds/

Another benefit is that if you're on a Windows machine without access to a mac, you can use the free tier on EAS to build for iOS and download it on your iPhone.

Hope this answers your question and gets you on your way. Feel free to re-open if you have additional questions.

**Comment by mars1211** [comment]
No, your answer is incorrect. If you run this command, it will still request access to the network and require you to enter your account and password to it.

```
eas build --profile production --platform android --local
An Expo user account is required to proceed.

Log in to EAS with email or username (exit and run eas login --help to see other login options)
? Email or username › 

```

**Comment by frankcalise** [comment]
@mars1211 you can run prebuild and utilize the respective IDEs such as Xcode and Android studio to run your builds. You can read more about there here: https://docs.expo.dev/build-reference/local-builds/

Also, just because a paid service is an _option_ doesn't make it repulsive. As mentioned, Nick provided _one option_ that is a free way to build, which is obviously a concern of yours. There are other options, but his is not a wrong suggestion by any means.

You certainly don't have to use the cloud service. Keep in mind that Expo is the recommended framework to use even by Meta, so we're not totally out of line here. Your tone, however, is and you won't get much help in the future from others if you don't respect people trying to help you.

If you don't agree with the stack, that's perfectly fine - you can build your own.

**Comment by mars1211** [comment]
It's a pity that your colleague was offended that his statement is not true, which I have confirmed in the quote.

As I understand it, you can't do a fully local build with eas (with the --local flag specified)

It's a mystery to me why you didn't specify this in the documentation and can't give an answer in this thread.

According to the idea for android you can run the build like this:
```
"build:android:prod": "npx expo prebuild && cd android && ./gradlew assembleRelease",
```

But it ends with this conclusion:
```
> npx expo prebuild && cd android && ./gradlew assembleRelease

✔ Created native directories | reusing /android, /ios
› Using current versions instead of recommended react-native@0.73.6.
✔ Updated package.json | no changes
✔ Finished prebuild
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Settings file '/test-project/android/settings.gradle' line: 11

* What went wrong:
A problem occurred evaluating settings 'test-project'.
> Could not read script '/' as it is a directory.

```

**Reply by frankcalise** [reply]
I didn't say he was offended. I asked you to talk to people in a respectful manner.

This project doesn't exist to solve all of your problems. This repo represents how we typically do projects and the way you're looking to build is not that, that's why it isn't in the documentation.

The build is fully local, it is done on your hardware. Yes you need an EAS account to get the project ID, but that is it. It doesn't do the build on the cloud.

**Reply by morganick** [reply]
Not offended. Just saw your discussion and wanted to respond before I signed off for a long weekend with the fam. Looks like @frankcalise got you squared away.

However, your word choice comes off adversarial. We're sharing something we use with the community at large. There are trade-offs that we're going to make that may not be in alignment with your needs or desires. That's ok. Share with us why those trade-offs don't work for you and maybe we can make the project better for everyone.

**Comment by mars1211** [comment]
Don't get me wrong, you have made a cool project, it is really logical and pleasant. 
But to tie it to external services without specifying an alternative is a sign of bad tone. 
And the fact that only I pointed it out to you does not make it only my problem. Please take criticism adequately. I apologize if something sounded rude. 
Thank you.

**Reply by frankcalise** [reply]
Understood, thanks for the feedback. We'll look into updating the documentation for that use case when we have time
